tests/test_hub_features.py

```diff
@@ -112,3 +113,65 @@ def test_load_different_peft_and_base_model_revision(self, tmp_path):
 
         assert peft_model.peft_config["default"].base_model_name_or_path == base_model_id
         assert peft_model.peft_config["default"].revision == base_model_revision
+
+
+class TestModelCard:
+    @pytest.mark.parametrize(
+        "model_id, peft_config, tags, excluded_tags",
+        [
+            (
+                "hf-internal-testing/tiny-random-Gemma3ForCausalLM",
+                LoraConfig(),
+                ["peft:method:lora", "base_model:adapter:hf-internal-testing/tiny-random-Gemma3ForCausalLM", "lora"],
+                [],
+            ),
+            (
+                "hf-internal-testing/tiny-random-Gemma3ForCausalLM",
+                BoneConfig(),
+                ["peft:method:bone", "base_model:adapter:hf-internal-testing/tiny-random-Gemma3ForCausalLM"],
+                ["lora"],
+            ),
+            (
+                "hf-internal-testing/tiny-random-BartForConditionalGeneration",
+                LoraConfig(),
+                [
+                    "peft:method:lora",
+                    "base_model:adapter:hf-internal-testing/tiny-random-BartForConditionalGeneration",
+                    "lora",
+                ],
+                [],
+            ),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "pre_tags",
+        [
+            ["tag1", "tag2"],
+            [],
+        ],
+    )
+    def test_model_card_has_expected_tags(self, model_id, peft_config, tags, excluded_tags, pre_tags, tmp_path):
+        """Make sure that PEFT sets the tags in the model card automatically and correctly.
+        This is important so that a) the models are searchable on the Hub and also 2) some features depend on it to
+        decide how to deal with them (e.g., inference).
+
+        Makes sure that the base model tags are still present (if there are any).
+        """
+        base_model = AutoModelForCausalLM.from_pretrained(model_id)
+
+        if pre_tags:
+            base_model.add_model_tags(pre_tags)
+
+        peft_model = get_peft_model(base_model, peft_config)
+        save_path = tmp_path / "adapter"
+
+        peft_model.save_pretrained(save_path)
+
+        model_card = ModelCard.load(save_path / "README.md")
+        assert set(tags).issubset(set(model_card.data.tags))
+
+        if excluded_tags:
+            assert not set(excluded_tags).issubset(set(model_card.data.tags))
```

Comment:
I wonder if this should be:

```suggestion
            assert set(excluded_tags).isdisjoint(set(model_card.data.tags))
```

The issue with `issubset` is that it requires each item to be in the 2nd set. Thus if `excluded_tags = {"a", "b"}` and `model_card.data.tags = {"b", "c"}`, the assertion would not fail.